libraries/custom_appbase/tests/custom_appbase_tests.cpp

```diff
@@ -350,7 +350,7 @@ BOOST_AUTO_TEST_CASE( execute_many_from_read_only_and_read_exclusive_queues ) {
    app->executor().set_to_read_window([](){return false;});
 
    // post functions
-   constexpr size_t num_expected = 600u;
+   constexpr int num_expected = 600;
```

Comment:
Wouldn't this change cause a warning in `for (size_t i = 400; i < num_expected; i+=5)` below?
I think it would make sense to cast the result of `std::count_if` to `size_t`. The result will always be positive, and the only reason it returns a signed value is for compatibility with `std::count` I believe.

Reply:
makes it clean on gcc 12.2 :woman_shrugging:  not sure what exactly the rules are: the warning is
> when a comparison between signed and unsigned values could produce an incorrect result when the signed value is converted to unsigned

So maybe there are additional rules here beyond just simply two types differentiating in signness which are required to be met before the warning is flagged?